### PR TITLE
EOS-24033: Optional/Mandatory parameter separation

### PR DESCRIPTION
**Pull request by pranavPawar07**
Signed-off-by: Pranav Diliprao Pawar <pranav.pawar@seagate.com>

# Problem Statement
- Optional/Mandatory parameter separation

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by pranavPawar07** [comment]
Cortx deployment : http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Cortx-Deployment/job/CentOS-7.9.2009/job/s3server/70/console

File: scripts/provisioning/setupcmd.py [hunk 1]
```diff
@@ -118,14 +118,21 @@ def get_confvalue(self, key: str):
     assert self.provisioner_confstore != None
     return self.provisioner_confstore.get_config(key)
 
+  def get_confvalue_with_validation(self, key: str):
```
Comment (nileshgovande) [inline scripts/provisioning/setupcmd.py:121]: With_defaults

File: scripts/provisioning/setupcmd.py [hunk 2]
```diff
@@ -118,14 +118,21 @@ def get_confvalue(self, key: str):
...
+    assert self.provisioner_confstore != None
+    value = self.provisioner_confstore.get_config(self.get_confkey(key))
+    if value is None:
+      value = self.s3_confkeys_store.get_config("DEFAULT_"+ key)
```
Comment (nileshgovande) [inline scripts/provisioning/setupcmd.py:125]: Use try construct and check for None value
1)if exception happens from lower layer or if value is None, throw exception here itself.

File: scripts/provisioning/setupcmd.py [hunk 3]
```diff
@@ -118,15 +117,32 @@ def get_confvalue(self, key: str):
     assert self.provisioner_confstore != None
     return self.provisioner_confstore.get_config(key)
 
+  def get_confvalue_with_defaults(self, key: str):
+    assert self.provisioner_confstore != None
+    if "DEFAULT" in key:
```
Comment (nileshgovande) [inline scripts/provisioning/setupcmd.py:122]: Check for beginning and use 'DEFAULT_'

**Pull request merged**